src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1374,11 +1374,6 @@ private function registerRouterConfiguration(array $config, ContainerBuilder $co
             $container->getDefinition('routing.loader')->replaceArgument(1, ['utf8' => true]);
         }
 
-        if ($enabledLocales) {
-            $enabledLocales = implode('|', array_map('preg_quote', $enabledLocales));
-            $container->getDefinition('routing.loader')->replaceArgument(2, ['_locale' => $enabledLocales]);
```

Comment:
Here, we expect the `_locale` value to be a string with all locale codes escaped and separated by a `|` (a regex).

An idea is that the config can accept a string and pass it as-this to the `routing.loader` service.

Alternatively we can modify the loader class and accept an array, then it could generate the regex.